stateful-functions-flink/stateful-functions-flink-core/src/main/java/com/ververica/statefun/flink/core/message/MessagePayloadSerializerMultiLanguage.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Ververica GmbH.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.statefun.flink.core.message;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.DynamicMessage;
+import com.ververica.statefun.flink.core.generated.Payload;
+import javax.annotation.Nonnull;
+
+public final class MessagePayloadSerializerMultiLanguage implements MessagePayloadSerializer {
+
+  @Override
+  public Object deserialize(@Nonnull ClassLoader targetClassLoader, @Nonnull Payload payload) {
+    return Any.newBuilder().setTypeUrl(payload.getClassName()).setValue(payload.getPayloadBytes());
+  }
+
+  @Override
+  public Payload serialize(@Nonnull Object what) {
+    final Any any;
+    if (what instanceof DynamicMessage) {
```

Comment:
Just a question to make sure I understood this correctly:
Within the stateful functions job, with `WITH_PROTOBUF_MULTILANG`, the message payloads are always either a `DynamicMessage` or a `Any`.
The only place where they would be `DynamicMessage`s are when they leave the `IngressRouterFlatMap`, correct?

Reply:
If the above is correct, I wonder if the packing of `DynamicMessage` to an `Any` should happen within the `IngressRouterFlatMap`, and this serializer only take cares of `Any`s?

Reply:
Your understanding is correct.
However the IngressFlatMap works with other payloads as well

Reply:
I see, and good point.
I'm ok with this as the first version, but eventually it would be nice to separately implement this as 2 different serializers, one for `DynamicMessage` and one for `Any`, as hinted by always checking `instanceOf` in serialize / copy.